src/PIL/ImageGrab.py

```diff
@@ -61,7 +61,17 @@ def grab(bbox=None, include_layered_windows=False, all_screens=False, xdisplay=N
                 left, top, right, bottom = bbox
                 im = im.crop((left - x0, top - y0, right - x0, bottom - y0))
             return im
-        elif shutil.which("gnome-screenshot"):
+    try:
+        if not Image.core.HAVE_XCB:
+            msg = "Pillow was built without XCB support"
+            raise OSError(msg)
+        size, data = Image.core.grabscreen_x11(xdisplay)
+        im = Image.frombytes("RGB", size, data, "raw", "BGRX", size[0] * 4, 1)
+        if bbox:
+            im = im.crop(bbox)
+        return im
+    except OSError:
+        if xdisplay is None and shutil.which("gnome-screenshot"):
```

Comment:
```suggestion
        if (
            sys.platform not in ("darwin", "win32")
            and xdisplay is None
            and shutil.which("gnome-screenshot")
        ):
```
To at least be clear in the code that we're expecting gnome-screenshot on macOS and Windows?

Reply:
Sure. How about ordering them same as the checks above?

```suggestion
        if (
	            xdisplay is None
	            and sys.platform not in ("darwin", "win32")
	            and shutil.which("gnome-screenshot")
	        ):
```

Reply:
Sure.